ingestion-lambda/src/handler.ts

```diff
@@ -1,83 +1,86 @@
 import { PutObjectCommand } from '@aws-sdk/client-s3';
-import type { SQSBatchResponse, SQSEvent, SQSRecord } from 'aws-lambda';
+import type { SQSBatchResponse, SQSEvent } from 'aws-lambda';
 import { BUCKET_NAME } from './config';
+import { tableName } from './database';
+import { createDbConnection } from './rds';
 import { s3Client } from './s3';
 
+type OperationOutcome = 'success' | 'failure';
+
+interface OperationResult {
+	sqsMessageId: string;
+	status: OperationOutcome;
+}
+
 export const main = async (event: SQSEvent): Promise<SQSBatchResponse> => {
 	const records = event.Records;
 
+	const sql = await createDbConnection();
+
 	console.log(`Processing ${records.length} messages`);
 
-	const eventBodies = records.map((record: SQSRecord) => ({
-		sqsMessageId: record.messageId,
-		snsMessageContent: record.body,
-		messageAttributes: record.messageAttributes,
-	}));
+	const results = await Promise.all(
+		records.map(
+			async ({
+				messageId,
+				messageAttributes,
+				body,
+			}): Promise<OperationResult> => {
+				const sqsMessageId = messageId;
 
-	const responses = await Promise.all(
-		eventBodies.map(
-			({ sqsMessageId: messageId, messageAttributes, snsMessageContent }) => {
-				const fingerpostMessageId =
-					messageAttributes['Message-Id']?.stringValue;
+				try {
```

Comment:
`try {} catch {}` around the whole block doesn't feel _ideal_, but it felt like it made sense for now given that we're not actually sure yet how we want to handle the potential dependencies between the different operations

Reply:
could you change to [`Promise.allSettled`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) and allow the async block to throw (which gets turned into promise rejection) then iterate, to log any failures.